code/obj/item/implant.dm

```diff
@@ -890,6 +896,366 @@ ABSTRACT_TYPE(/obj/item/implant/revenge)
+			ping_reply.source = src
+			ping_reply.data["device"] = "IMP_MARIONETTE"
+			ping_reply.data["sender"] = src.net_id
+			ping_reply.data["address_1"] = signal.data["sender"]
+			ping_reply.data["command"] = "ping_reply"
+			ping_reply.data["passkey"] = src.passkey
+			ping_reply.data["status"] = src.heat > src.heat_danger_zone ? "DANGER" : ismob(src.owner) ? "ACTIVE" : "IDLE"
+			SEND_SIGNAL(src, COMSIG_MOVABLE_POST_RADIO_PACKET, ping_reply)
+			return
+
+		if (src.passkey && src.passkey != signal.data["passkey"])
+			if (signal.data["sender"] != src.linked_address)
+				send_activation_reply(signal.data["sender"], MARIONETTE_IMPLANT_ERROR_BAD_PASSKEY)
+				return
+
+		src.heat_dissipation = initial(src.heat_dissipation)
+
+		var/fail_reason
+		if (!ismob(src.owner))
+			fail_reason = MARIONETTE_IMPLANT_ERROR_NO_TARGET
+			send_activation_reply(signal.data["sender"], fail_reason)
+			return
+
+		var/mob/living/carbon/human/H = src.owner
+		if (istype(H) && H.decomp_stage != DECOMP_STAGE_NO_ROT)
+			fail_reason = MARIONETTE_IMPLANT_ERROR_DEAD_TARGET
+			send_activation_reply(signal.data["sender"], fail_reason)
+			return
+
+		var/data = signal.data["data"]
+		switch (command)
+			if ("say", "speak")
+				if (!isdead(src.owner))
+					logTheThing(LOG_COMBAT, src.owner, "was forced by \a [src] to say \"[data]\" at [log_loc(src.owner)] (caused by [constructTarget(signal.author, "combat")] at [log_loc(signal.author)]).")
+					data = copytext(strip_prefix(data, "*"), 1, 46) // Trim starting asterisks to prevent force-emoting
+					src.owner.say(data)
+				else
+					fail_reason = MARIONETTE_IMPLANT_ERROR_DEAD_TARGET
+				src.adjust_heat(15)
+			if ("emote")
+				data = lowertext(data)
+				if (!isdead(src.owner))
+					if (data in src.emote_blacklist)
+						fail_reason = MARIONETTE_IMPLANT_ERROR_INVALID
+					else
+						logTheThing(LOG_COMBAT, src.owner, "was forced by \a [src] to emote \"[data]\" at [log_loc(src.owner)] by (caused by [constructTarget(signal.author, "combat")] at [log_loc(signal.author)]).")
+						src.owner.emote(data)
+				else
+					fail_reason = MARIONETTE_IMPLANT_ERROR_DEAD_TARGET
+				src.adjust_heat(15)
+			if ("move", "step", "bump")
+				logTheThing(LOG_COMBAT, src.owner, "was forced by \a [src] to step to the [lowertext(data)] at [log_loc(src.owner)] (caused by [constructTarget(signal.author, "combat")] at [log_loc(signal.author)]).")
+				var/step_dir = text2dir(uppertext(data))
+				if (step_dir && (step_dir in cardinal))
+					step(src.owner, step_dir)
+				else
+					fail_reason = MARIONETTE_IMPLANT_ERROR_INVALID
+				src.adjust_heat(5)
+			if ("shock", "zap")
+				logTheThing(LOG_COMBAT, src.owner, "was shocked by \a [src] at [log_loc(src.owner)] (caused by [constructTarget(signal.author, "combat")] at [log_loc(signal.author)]).")
+				boutput(src.owner, SPAN_ALERT("You feel a shock from inside your body!"))
+				src.owner.do_disorient(90, knockdown = 7 SECONDS, disorient = 3 SECONDS)
```

Comment:
This doesn't seem to do what you think it does.

Reply:
the intent is that it chunks for 90 stamina damage, knocks down for 7 seconds if it drains stamina completely, and disorients for 3 seconds regardless. is that not what it's doin' here?

Reply:
Seems to just be doing the stamina damage with no disorient

Reply:
huh, weird. I recall it working during testing, and I can actually see the disorient effect working in the demo videos I recorded. new bug that got introduced or something? if I can repro it locally after updating to the latest master I'll comment here

Reply:
so I wasn't actually able to reproduce this on my end while on the devtest map -- the shock correctly caused the disorient effect when I wasn't being knocked down outright by stamina loss. under what conditions did it not work right for you? I'll try and repro them locally

Reply:
Seems to work now, not sure if you changed something or I was just being dumb!